scripts/dump-co-pilot-reporting-data.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+set -euo pipefail
+
+if ! command -v db-to-sqlite &>/dev/null; then
+    echo 'db-to-sqlite was not found'
+    exit 1
+fi
+
+if [ -z ${DATABASE_URL+none} ]; then
```

Comment:
What does `+none` do here?

Reply:
If `DATABASE_URL` isn't set, then the braces will evaluate to an empty string. If it is set, then the braces will evaluate to the string `none`. In the former case, [the condition will be true][1] (length is zero). In the latter case, the condition will be false (length is four). I couldn't think of a better token. Possibly `false`? (More on [SO][2].)

Why bother? Well, without the braces, if `DATABASE_URL` isn't set, then the script terminates with

```
./scripts/dump-co-pilot-reporting-data.sh: line 9: DATABASE_URL: unbound variable
```

is displayed, because of `set -eu`.

[1]: https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html
[2]: https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash

Reply:
Well that's no fun, is it?  @bloodearnest – is there a better way you know of to check for an env var existing and also having a value?

Reply:
I could totally do a yarn on test vs [ vs [[, thinking about it.

Reply:
I'd love a yarn on that! How about Thursday 25th January?

Reply:
Thank you for your comments, @bloodearnest. Really useful and really interesting! I've updated the script.